pkg/lib/cockpit.d.ts

```diff
@@ -93,6 +93,49 @@ declare module 'cockpit' {
     function channel(options: ChannelOptions & { binary?: false; }): Channel<string>;
     function channel(options: ChannelOptions & { binary: true; }): Channel<Uint8Array>;
 
+    /* === cockpit.spawn ============================= */
+
+    interface DeferredPromise<T> extends Promise<T> {
+        /* jQuery Promise compatibility */
+        done(callback: (data: T) => void): DeferredPromise<T>
+        fail(callback: (exc: Error) => void): DeferredPromise<T>
+        always(callback: () => void): DeferredPromise<T>
+        progress(callback: (message: T, cancel: () => void) => void): DeferredPromise<T>
+    }
+
+    interface Deferred<T> {
+        resolve(): Deferred<T>;
+        reject(): Deferred<T>;
+        notify(): Deferred<T>;
+        promise: DeferredPromise<T>
+    }
+
+    interface Spawn<T> extends DeferredPromise<T> {
+        input(message: T, stream?: boolean): DeferredPromise<T>;
+        stream(callback: (data: T) => void): DeferredPromise<T>;
+        close(): void;
+    }
+
+    interface SpawnOptions {
+        binary?: boolean,
```

Comment:
I'm not sure how I feel about listing `binary` here, since we discriminate on it in the function definitions.  It's harmless, but it also doesn't bring anything...